go/mysql/sql_error.go

```diff
@@ -120,7 +120,7 @@ func NewSQLErrorFromError(err error) error {
 		case vtrpcpb.Code_UNAUTHENTICATED:
 			num = ERAccessDeniedError
 		case vtrpcpb.Code_RESOURCE_EXHAUSTED:
-			num = ERTooManyUserConnections
+			num = demuxResourceExhaustedErrors(err.Error())
```

Comment:
is this still required after changing to through mysql error directly?

Reply:
Yes -- if we directly map all ResourceExhausted into ERNetPacketTooLarge then we end up in the same position where we are grouping errors that don't necessarily make sense together.

Specifically when we overflow a connection pool (`vttablet: rpc error: code = ResourceExhausted desc = pool ConnPool waiter count exceeded`) and the like we want to distinguish that as something that can be solved via backoff instead of queries that are inherently less retryable because they are likely to generate "bad" data.